### PR TITLE
Qtpydocking

### PR DESCRIPTION
**Pull request by teddyrendahl**
Two fewer lines of  library code and so much more feature! Nice work @klauer

### CODE REVIEW

File: lucid/main_window.py [hunk 1]
```diff
@@ -30,57 +30,53 @@ class LucidMainWindow(QMainWindow):
...
+        self.main_dock.setWidget(self.dock_manager)
+        self.main_dock.installEventFilter(self)
+        # Add to the first allowed location
+        self.addDockWidget(self.allowed_docks[0], self.main_dock)
+        return self.main_dock
+
+    def eventFilter(self, obj, event):
+        """Qt eventFilter to watch for closures of the docking system"""
+        # If the user closes the docking system clean up our internal state
+        if (self.main_dock
```
Comment (klauer) [inline lucid/main_window.py]: Hmm, I think I should try to do better on the qtpydocking side, emitting a signal on close. While I don't have numbers, I'm fairly certan qt event filters are very pricey in Python.

The `if` block indentation looks off here, by the way...

File: requirements.txt [hunk 2]
```diff
@@ -0,0 +1,4 @@
+pydm
+git+https://github.com/klauer/qtpydocking
+qtpy
+
```
Comment (klauer) [inline requirements.txt]: Blank line

File: lucid/main_window.py (hunk 1 above)
Comment (teddyrendahl) [reply lucid/main_window.py]: 1. Hadn't heard about the cost of the `eventFilter` before, makes sense though if every event has to go through the filter... What's tricky about this (and this code in general), is this is a `PyQt.QDockWidget` not a `qtpydocking.DockWidget`, this block is watching for closing the whole dock system (and firing a reset)

Comment (teddyrendahl) [reply lucid/main_window.py]: Alternative method in https://github.com/pcdshub/lucid/pull/2/commits/da5bfcf2a9b95079dd63e360eb9ec0c9fc4b870d

Comment (teddyrendahl) [reply lucid/main_window.py]: Alternative method in https://github.com/pcdshub/lucid/pull/2/commits/da5bfcf2a9b95079dd63e360eb9ec0c9fc4b870d

**Comment by teddyrendahl** [comment]
@klauer we aren't going to find bugs not using it!

**Pull request merged**